doc/manual/merging.md

```diff
@@ -0,0 +1,522 @@
+# Merging records
+
+In Nickel, the basic building blocks for data are records (objects in JSON or
+attribute sets in Nix). Merging is a fundamental built-in operation whose role
+is to combine records togethers. It is useful to compose small and logical
+blocks into a potentially complex final configuration, making it more
+manageable. Merge is performed by the `&` operator.
+
+Merge is a **symmetric** operation (or, pedantically, commutative). In practice,
+this means that order doesn't matter, and `left & right` is the same thing as
+`right & left`. When the operands need to be distinguished, as we will see for
+default values for example, the idea is to use metadata to do so (annotations),
+rather than relying on the left or right position.
+
+**Warning**: At the time of writing, Nickel's version is 0.1. Important
+additions to merging are planned for coming versions, including priorities and
+custom merge functions. They are not detailed here yet. For more details, see
+the associated technical document [RFC001][rfc001].
+
+The section describes the behavior and use-cases of merge, by considering the
+following situations:
+
+- Merging two records without common fields
+- Merging records with common fields
+- Merging records with metadata
+  * Default values
+  * Contracts
+  * Documentation
+- Recursive overriding
```

Comment:
Could these link to the relevant sections (and maybe make it more explicit that it’s a toc, I didn’t notice it at first)?